internal/plugins/src/main/kotlin/org/jsonschema2dataclass/internal/plugin/publishing/publishing.kt

```diff
@@ -0,0 +1,116 @@
+package org.jsonschema2dataclass.internal.plugin.publishing
+
+import basePluginExtension
+import groovy.util.Node
+import groovy.util.NodeList
+import org.gradle.api.Project
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.kotlin.dsl.configure
+import org.gradle.kotlin.dsl.get
+import org.gradle.kotlin.dsl.getByName
+import org.gradle.kotlin.dsl.register
+import org.gradle.plugins.signing.SigningExtension
+
+fun applyPublishing(project: Project, signing: Boolean = false) {
+    project.plugins.apply("maven-publish")
+
+    project.configure<PublishingExtension> {
+        publications {
+            register<MavenPublication>("release") release@{
```

Comment:
> Do you have a github action for an actual publication to Nexus?

Yes and no.

Here's the release script:
https://github.com/TWiStErRob/net.twisterrob.gradle/blob/master/docs/release.md

BUT as part of the CI, I publish every PR/build to sonatype snapshot from here:
https://github.com/TWiStErRob/net.twisterrob.gradle/blob/f738274f390c752db8d1406b96b70738adbcb7d9/.github/workflows/CI.yml#L146-L161
and then immediately use it:
https://github.com/TWiStErRob/net.twisterrob.gradle/blob/f738274f390c752db8d1406b96b70738adbcb7d9/.github/workflows/CI.yml#L163-L169
so it should be easy to repurpose this as a workflow_dispatch manual action. It's on my todo list :D

The publishing uses `<version>.<branch>-SNAPSHOT` pattern, see existing builds for the entry points:
https://github.com/TWiStErRob/net.twisterrob.gradle/blob/master/docs/development.md#using-the--snapshot-from-sonatype

Reply:
I'll do the similar publishing for this plugin in a next PR

Reply:
I have all secrets set only in `production` environment, normal build don't use any secrets by default, and preferably checkout is as shallow as possible (so no tags as well). This is one more reason to use local maven repository

Environment documentation
https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment